types/types.go

```diff
@@ -23,67 +24,158 @@ import (
 	"github.com/prometheus/common/model"
 )
 
+type state uint8
+
+const (
+	Unprocessed state = iota
```

Comment:
I'm not sure if `Unprocessed` is the best name for it, I was thinking about calling it `New` but that on the other hand might be mistaken for an alert that _just fired_ rather than indicate what we really mean here.